Add configuration option to disable repeating of prior command

Copyright (C) 2013 Curt Brune <curt@cumulusnetworks.com>
Copyright (C) 2015 david_yang <david_yang@accton.com>

SPDX-License-Identifier:     GPL-2.0

Prior to history and an editable command line, u-boot had a "feature" that
allowed users to type CR at the prompt and repeat the last command (some
commands even auto-increment addresses).

There are emails on the mailing list suggesting that the fucntion be limited to
"useful" cases; however, it is clear that this "repeat" leads to unexpected
behavior as it differs from almost every standard command parser.

This patch allows users to completely disable the repeat to be completely
disabled by defining "CONFIG_SYS_NO_REPEATABLE.

diff --git a/common/cli_hush.c b/common/cli_hush.c
index 38da5a0..a94d4c6 100644
--- a/common/cli_hush.c
+++ b/common/cli_hush.c
@@ -1027,6 +1027,9 @@ static void get_user_input(struct in_str *i)
 	if (had_ctrlc()) flag_repeat = 0;
 	clear_ctrlc();
 	do_repeat = 0;
+#ifdef CONFIG_SYS_NO_REPEATABLE
+	flag_repeat = 0;
+#endif
 	if (i->promptmode == 1) {
 		if (console_buffer[0] == '\n'&& flag_repeat == 0) {
 			strcpy(the_command,console_buffer);
diff --git a/common/cli_simple.c b/common/cli_simple.c
index 353ceeb..dd1f94e 100644
--- a/common/cli_simple.c
+++ b/common/cli_simple.c
@@ -248,6 +248,9 @@ int cli_simple_run_command(const char *cmd, int flag)
 		if (cmd_process(flag, argc, argv, &repeatable, NULL))
 			rc = -1;
 
+#ifdef CONFIG_SYS_NO_REPEATABLE
+		repeatable = 0;
+#endif
 		/* Did the user stop this? */
 		if (had_ctrlc())
 			return -1;	/* if stopped then not repeatable */
